### PR TITLE
Add docs for VSchema DDL

### PR DESCRIPTION
**Pull request by davidpiegza**
Add docs for VSchema DDL.

A new page has been added to `User Guides -> VSchema and Query Serving -> VSchema DDL`. The content is taken from https://github.com/vitessio/vitess/blob/6d26dad29110566ded746f5013bd1115d9f200b4/examples/demo/vschema_ddls.sql.

Please note, this is a first version and probably not complete. I added it for V17.0 only and would copy it to other versions after first reviews.

- Related issue: https://github.com/vitessio/vitess/issues/13903

### CODE REVIEW

**Comment by deepthi** [comment]
@davidpiegza thank you for the contribution. Do you mind fixing the DCO by amending your commit with a `--signoff`?

File: content/en/docs/17.0/user-guides/vschema-guide/vschema_ddl.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+title: VSchema DDL
+weight: 30
+---
+
+VSchema DDL is an experimental feature that allows users to alter the VSchema by issuing "vschema ddls" directly to vtgate. The vschema_ddl_authorized_users specifies which users can alter the vschema.
```
Comment (deepthi) [inline content/en/docs/17.0/user-guides/vschema-guide/vschema_ddl.md]:
```suggestion
VSchema DDL is an experimental feature that allows users to alter the VSchema by issuing "vschema ddls" directly to vtgate. The `vschema_ddl_authorized_users` flag specifies which users can alter the vschema.
```

**Comment by davidpiegza** [comment]
@deepthi I applied the suggestions and referenced the new page in a vtgate doc. If this looks good I can add the documentation to the other versions.

File: content/en/docs/17.0/user-guides/vschema-guide/vschema_ddl.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+### Lookup as Primary Vindex
+
+```sql
+alter vschema add sequence product.corder_event_seq;
+alter vschema on customer.corder_event add vindex corder_keyspace_idx(corder_id);
+alter vschema on customer.corder_event add auto_increment corder_event_id using product.corder_event_seq;
+
+-- Reversible Vindexes
+alter vschema on customer.corder_event add vindex `binary`(keyspace_id) using `binary`;
```
Comment (arthurschreiber) [inline content/en/docs/17.0/user-guides/vschema-guide/vschema_ddl.md]: Won't the execution of the last ddl here change the lookup vindex to no longer be the primary vindex? 🤔

File: content/en/docs/17.0/user-guides/vschema-guide/vschema_ddl.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+alter vschema add sequence product.customer_seq;
+alter vschema on customer.customer add auto_increment customer_id using product.customer_seq;
+
+-- Drop sequence and auto_increment
+alter vschema drop sequence product.customer_seq;
+alter vschema on customer.customer drop auto_increment;
+```
+
+
+### Shared Vindexes and Foreign Keys
```
Comment (arthurschreiber) [inline content/en/docs/17.0/user-guides/vschema-guide/vschema_ddl.md]: This title does not really match the ddls that are described here, does it?

Comment (deepthi) [reply content/en/docs/17.0/user-guides/vschema-guide/vschema_ddl.md]: Actually I think it does. But for it to be easy to understand, we should show that `customer.customer` is also being sharded on `customer_id`.

**Review by frouioui — COMMENTED** [review]
Unless we backport https://github.com/vitessio/vitess/pull/13882 to `release-17.0`, let's move all of this documentation to `content/en/docs/18.0/...` please

**Comment by davidpiegza** [comment]
@frouioui @deepthi I moved it to v18.0 docs and also decided with @arthurschreiber to change the page to actually describe each query on it's own (and not use the examples from the vitess repo).

Here is a [readable version](https://github.com/vitessio/website/blob/115f040d3fd5efb2e119e2091a5046e46c445c6a/content/en/docs/18.0/user-guides/vschema-guide/vschema_ddl.md).

**Review by frouioui — APPROVED** [review]

**Comment by frouioui** [comment]
Waiting for https://github.com/vitessio/vitess/pull/13882 to be merged before merging this PR.

**Pull request merged**